crawl-ref/source/beam.cc

```diff
@@ -1533,14 +1509,13 @@ int mons_adjust_flavoured(monster* mons, bolt &pbolt, int hurted,
     case BEAM_HOLY:
     {
         hurted = resist_adjust_damage(mons, pbolt.flavour, hurted);
-        if (doFlavouredEffects && (!hurted || hurted != original))
+        if (doFlavouredEffects && original > 0
+            && (!hurted || hurted != original))
         {
-            simple_monster_message(*mons,
-                                       original == 0 ? " appears unharmed." :
-                                         hurted == 0 ? " completely resists." :
-                                   hurted < original ? " resists." :
-                                   hurted > original ? " writhes in agony!" :
-                                   " suffers an impossible damage scaling amount!");
+            simple_monster_message(*mons, hurted == 0 ? " completely resists." :
+                                    hurted < original ? " resists." :
+                                    hurted > original ? " writhes in agony!" :
+                               " suffers an impossible damage scaling amount!");
```

Comment:
Any plausible modern compiler should be able to prove that this last case doesn't happen; better to delete it.